source/fem-dem/ib_particles_dem.cc

```diff
@@ -464,175 +528,291 @@ IBParticlesDEM<dim>::calculate_pw_contact_force(
-              pw_contact_map[particle.particle_id][boundary_index]
-                .normal_vector = normal;
-              pw_contact_map[particle.particle_id][boundary_index].boundary_id =
-                0;
-
-              particle_wall_contact_force_object
-                ->calculate_IB_particle_wall_contact_force(
-                  pw_contact_map[particle.particle_id][boundary_index],
-                  normal_force,
-                  tangential_force,
-                  tangential_torque,
-                  rolling_resistance_torque,
-                  particle,
-                  wall_youngs_modulus,
-                  wall_poisson_ratio,
-                  wall_restitution_coefficient,
-                  wall_friction_coefficient,
-                  wall_rolling_friction_coefficient,
-                  dt_dem,
-                  particle.mass,
-                  particle.radius);
-              // Updating the force of particles in the particle handler
-              contact_force[particle.particle_id] -=
-                (normal_force + tangential_force);
-              // Updating the torque acting on particles
-              contact_torque[particle.particle_id] +=
-                tangential_torque + rolling_resistance_torque;
-            }
-          else
-            {
-              // Set to 0 the tangential overlap if the particle is not in
-              // contact with the wall anymore.
-              for (int d = 0; d < dim; ++d)
+                  Tensor<1, 3> normal_force;
+                  Tensor<1, 3> tangential_force;
+                  Tensor<1, 3> tangential_torque;
+                  Tensor<1, 3> rolling_resistance_torque;
+                  Tensor<1, 3> wall_velocity;
+                  Tensor<1, 3> wall_angular_velocity;
+                  Point<dim>   wall_center_of_rotation;
+
+                  pw_contact_map[particle.particle_id][boundary_index]
+                    .normal_overlap = normal_overlap;
+                  pw_contact_map[particle.particle_id][boundary_index]
+                    .normal_vector = normal;
+
+                  ObjectProperties particle_properties;
+                  particle_properties.object_friction_coefficient =
+                    particle.friction_coefficient;
+                  particle_properties.object_mass = particle.mass;
+                  particle_properties.object_poisson_ratio =
+                    particle.poisson_ratio;
+                  particle_properties.object_radius =
+                    contact_radius_particle_one;
+                  particle_properties.object_restitution_coefficient =
+                    particle.restitution_coefficient;
+                  particle_properties.object_rolling_friction_coefficient =
+                    particle.rolling_friction_coefficient;
+                  particle_properties.object_youngs_modulus =
+                    particle.youngs_modulus;
+
+                  // Wall mass is 100 times higher than
```

Comment:
I'm just wondering why the wall properties should depend on the properties of the particle contacting it. And why the factor 100, is it an arbitrary choice or motivated by something?

Reply:
the idea here is to make the wall way bigger than the particle while avoiding using inf. a 100 is an arbitrary choice but it is good enough.